### DISCUSSION TITLE
Plans for support of Parcelize for Darwin/Apple targets

### DISCUSSION THREAD

**Discussion by malliaridis** [arkivanov/Essenty#43] [General]
With the support of iOS and macOS native expected from https://github.com/arkivanov/Decompose/issues/74 it is probably a good time to introduce support for Darwin/Apple targets for Parcelize in this library, so the complete suite of Essenty / MVI-Kotlin / Decompose can be directly be used without (I guess) many code changes.

Are there any plans yet for bringing the proof of concept project [kotlin-parcelize-darwin](https://github.com/arkivanov/kotlin-parcelize-darwin) into this one?

**Comment by arkivanov** [comment]
Thanks for raising! Currently there are now plans. It is not clear how the benefits will outweigh the efforts. Essenty and Decompose are already being used in iOS and other Darwin platforms. It feels that process deaths happen way less often than on Android, and the state restoration is less important.

Another concern is the stability of the compiler plugin API. There is little to no documentation, and the API often changes. It requires a lot of effort to maintain custom compiler plugins.

So yeah, let's keep this discussion open. I hope to receive more feedback from the community.

**Reply by malliaridis** [reply]
Does that mean configuration changes that require state preservation on Android (for example screen orientation) do not affect iOS devices the same way like they do on Android? I'm not familiar yet with the iOS environment.

**Reply by arkivanov** [reply]
AFAIK there is no such thing as configuration change on iOS. There are APIs to preserve the state over the process's death, but from what I heard - it is rare and apps just start from the beginning.

**Comment by arkivanov** [comment]
Version [0.7.0](https://github.com/arkivanov/Essenty/releases/tag/0.7.0) adds an experimental support of Parcelize on all Darwin (Apple) targets!
